ruby/red-arrow/test/test-tensor.rb

```diff
@@ -16,6 +16,103 @@
+                       shape: @shape,
+                       strides: @strides,
+                       dimension_names: ["", "", ""],
+                     },
+                     {
+                       value_data_type: tensor.value_data_type,
+                       buffer: tensor.buffer.data.to_s,
+                       shape: tensor.shape,
+                       strides: tensor.strides,
+                       dimension_names: tensor.dimension_names,
+                     })
+      end
+
+      test("Array, data_type: Symbol") do
+        tensor = Arrow::Tensor.new(@raw_tensor, data_type: :int32)
+        assert_equal({
+                       value_data_type: Arrow::Int32DataType.new,
+                       buffer: @raw_tensor.flatten.pack("l*"),
+                       shape: @shape,
+                       strides: @strides.collect {|x| x * 4},
+                       dimension_names: ["", "", ""],
+                     },
+                     {
+                       value_data_type: tensor.value_data_type,
+                       buffer: tensor.buffer.data.to_s,
+                       shape: tensor.shape,
+                       strides: tensor.strides,
+                       dimension_names: tensor.dimension_names,
+                     })
+      end
+
+      test("Array, dimension_names: Array<String>") do
+        tensor = Arrow::Tensor.new(@raw_tensor,
+                                   dimension_names: ["a", "b", "c"])
+        assert_equal({
+                       value_data_type: Arrow::UInt8DataType.new,
+                       buffer: @raw_tensor.flatten.pack("C*"),
+                       shape: @shape,
+                       strides: @strides,
+                       dimension_names: ["a", "b", "c"],
+                     },
+                     {
+                       value_data_type: tensor.value_data_type,
+                       buffer: tensor.buffer.data.to_s,
+                       shape: tensor.shape,
+                       strides: tensor.strides,
+                       dimension_names: tensor.dimension_names,
+                     })
+      end
+
+      test("Array, dimension_names: Array<Symbol>") do
+        tensor = Arrow::Tensor.new(@raw_tensor,
+                                   dimension_names: [:a, :b, :c])
+        assert_equal({
+                       value_data_type: Arrow::UInt8DataType.new,
+                       buffer: @raw_tensor.flatten.pack("C*"),
+                       shape: @shape,
+                       strides: @strides,
+                       dimension_names: ["a", "b", "c"],
+                     },
+                     {
+                       value_data_type: tensor.value_data_type,
+                       buffer: tensor.buffer.data.to_s,
+                       shape: tensor.shape,
+                       strides: tensor.strides,
+                       dimension_names: tensor.dimension_names,
+                     })
+      end
```

Comment:
I don't see a test passing a different `strides` explicitly. Can you add one? Or does it just not make sense to change the strides?

Reply:
Ah, I forgot to add it. `strides` is for passing `Arrow::Buffer` (not `Array`) as the first argument.
I'll add some tests for `Arrow::Buffer`.